python/lsst/meas/extensions/ngmix/LMSimpleShape.py

```diff
@@ -146,41 +153,41 @@ def measure(self, measRecord, exposure):
-            psfObs.set_gmix(gmix)
+            psfGMix = ngmix.gmix.GMix(len(shapeletFunctions), psfPars)
+        psfObs.set_gmix(psfGMix)
 
         #   Now create an obs for the galaxy itself, including the weight plane
         galArray = exposure.getMaskedImage().getImage().getArray()
-        gal_jacob = ngmix.UnitJacobian(row=(galArray.shape[1] - 1)/2.0, col=(galArray.shape[0] - 1)/2.0)
+        galShape = galArray.shape
+        galJacob = ngmix.UnitJacobian(row=galShape[1]/2.0, col=galShape[0]/2.0)
         variance = exposure.getMaskedImage().getVariance().getArray()
         gal_weight = 1/variance
-        obs = Observation(galArray, weight=gal_weight, jacobian=gal_jacob, psf=psfObs)
-
-        #   Run detection on the source image to get a guess of the object pars
-        #   This will give us a rough guess about the shape of the object and its centroid
-        threshold = lsst.afw.detection.Threshold(5.0, lsst.afw.detection.Threshold.STDEV)
-        fpSet = lsst.afw.detection.FootprintSet(exposure.getMaskedImage().getImage(), threshold)
-        if len(fpSet.getFootprints()) > 1:
-            raise RuntimeError("Threshold value results in multiple Footprints for a single object")
-        if len(fpSet.getFootprints()) == 0:
-            raise RuntimeError("Threshold value results in zero Footprints for object")
-        fp = fpSet.getFootprints()[0]
-        x = fp.getCentroid().getX() - exposure.getXY0().getX()
-        y = fp.getCentroid().getY() - exposure.getXY0().getY()
+        obs = Observation(galArray, weight=gal_weight, jacobian=galJacob, psf=psfObs)
+
+        fp = measRecord.getFootprint()
+
+        bbox = fp.getBBox()
+        ymin = bbox.getBeginY() - exposure.getXY0().getY()
+        ymax = bbox.getEndY() - exposure.getXY0().getY()
+        xmin = bbox.getBeginX() - exposure.getXY0().getX()
+        xmax = bbox.getEndX() - exposure.getXY0().getX()
+        flux = exposure.getMaskedImage().getImage().getArray()[ymin:ymax, xmin:xmax].sum()
+
         xx = fp.getShape().getIxx()
         yy = fp.getShape().getIyy()
         xy = fp.getShape().getIxy()
-        #    set the guess as [cen1, cen2, g1, g2, T, flux]
-        guess = (x, y, (xx-yy)/(xx+yy), 2*xy/(xx+yy), xx+yy, galArray[x, y])
 
         #   Now run the shape algorithm, using the config parameters
         lmPars = {'maxfev': self.config.maxfev, 'ftol': self.config.ftol, 'xtol': self.config.xtol}
-        fitter = LMSimple(obs, self.config.model, lm_pars=lmPars)
-        fitter.go(guess=guess)
+        maxPars = {'method':'lm', 'lm_pars':lmPars}
+        guesser = ngmix.guessers.TFluxGuesser(xx+yy, flux)
+        runner = ngmix.bootstrap.MaxRunner(obs, self.config.model, maxPars, guesser)
+        runner.go(ntry=4)
+        fitter = runner.fitter
         res = fitter.get_result()
 
-        #   Set the results, including the fit info returned by EMRunner
-        #   We only know about two EM errors.  Anything else is thrown as "unknown".
+
```

Comment:
Unnecessary extra newline: should only have two in a row between functions or classes.